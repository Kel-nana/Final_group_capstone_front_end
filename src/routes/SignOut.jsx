import React from 'react';
import { useNavigate } from 'react-router-dom';

function SignOut() {
  const navigate = useNavigate();

  const handleSignOut = async () => {
    try {
      const response = await fetch(
        'https://doctalk-r977.onrender.com//users/sign_out',
        {
          method: 'DELETE',
          headers: {
            'Content-Type': 'application/json',
            Authorization: localStorage.getItem('token'),
          },
        },
      );
      localStorage.removeItem('token');

      if (response.ok) {
        // console.log('Sign-out successful.');

        navigate('/');
      } else {
        // console.error('Sign-out failed.');
      }
    } catch (error) {
      // console.error('Error:', error);
    }
  };

  return (
    <li className="w-full text-center py-4 mb-4 hover:text-white hover:bg-[#97bf0f] transition-all">
      <button onClick={handleSignOut} type="submit">
        SIGN OUT
      </button>
    </li>
  );
}

export default SignOut;
