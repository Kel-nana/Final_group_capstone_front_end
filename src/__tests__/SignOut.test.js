import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SignOut from '../routes/SignOut';

describe('Sign out button', () => {
  it('Should display on  the page', () => {
    const { container } = render(
      <BrowserRouter>
        <SignOut />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
