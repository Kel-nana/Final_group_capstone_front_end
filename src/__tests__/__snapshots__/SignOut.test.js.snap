// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign out button Should display on  the page 1`] = `
<div>
  <div>
    <button
      type="submit"
    >
      Sign Out
    </button>
  </div>
</div>
`;
